tests/src/test/java/org/eclipse/hono/tests/registry/CredentialsAmqpIT.java

```diff
@@ -86,11 +91,32 @@ public static void prepareDeviceRegistry(final TestContext ctx) {
             .setHandler(ctx.asyncAssertSuccess(r -> {
                 credentialsClient = r;
             }));
+
+        registryHttpClient = new DeviceRegistryHttpClient(
+                vertx,
+                IntegrationTestSupport.HONO_DEVICEREGISTRY_HOST,
+                IntegrationTestSupport.HONO_DEVICEREGISTRY_HTTP_PORT);
+    }
+
+    /**
+     * Remove the fixture from the device registry if the test had set up any.
+     *
+     */
+    @After
+    public void cleanupDeviceRegistry(){
+        if (testData != null) {
+            registryHttpClient.removeCredentials(
```

Comment:
@jbtrystram you should handle the result of this call and await for its execution to avoid async problems. Take a look at https://github.com/eclipse/hono/blob/master/tests/src/test/java/org/eclipse/hono/tests/registry/CredentialsHttpIT.java#L104 for example on how to do it.

Reply:
thanks for the tip !